simple/stats/data.py

```diff
@@ -227,35 +230,62 @@ def triples(self) -> list[Triple]:
     return triples
 
 
+@dataclass
+class ObservationProperties:
+  unit: str = ""
+  scaling_factor: str = ""
+  measurement_method: str = ""
+  observation_period: str = ""
+  # All custom properties other than the standard ones above go in this field.
+  properties: dict[str, str] = field(default_factory=dict)
+
+  @classmethod
+  def new(cls: Self, all_properties: dict[str, str] = {}) -> Self:
+    unit = all_properties.get(sc.PREDICATE_UNIT, "")
+    scaling_factor = all_properties.get(sc.PREDICATE_SCALING_FACTOR, "")
+    measurement_method = all_properties.get(sc.PREDICATE_MEASUREMENT_METHOD, "")
+    observation_period = all_properties.get(sc.PREDICATE_OBSERVATION_PERIOD, "")
+    custom_properties = {
+        p: v
+        for p, v in all_properties.items()
+        if p not in sc.STANDARD_OBSERVATION_PROPERTIES
+    }
+    return cls(unit, scaling_factor, measurement_method, observation_period,
+               custom_properties)
+
+
 @dataclass
 class Observation:
   entity: str
   variable: str
   date: str
   value: str
   provenance: str
-  properties: dict[str, str] = field(default_factory=dict)
-
-  def __post_init__(self):
-    if not self.properties:
-      self.properties = {}
-    # Properties in the DB are stored as gzipped and base64 encoded strings.
-    # Convert it to json / dict so it is available as a dict in code.
-    elif isinstance(self.properties, str):
-      self.properties = base64_decode_and_gunzip_json(self.properties)
-
-  def properties_string(self) -> str:
-    if not self.properties:
-      return ""
-    return gzip_and_base64_encode_json(self.properties)
+  properties: ObservationProperties = field(
+      default_factory=ObservationProperties.new)
 
   def db_tuple(self):
     return (_strip_namespace(self.entity), _strip_namespace(self.variable),
             self.date, self.value, _strip_namespace(self.provenance),
-            self.properties_string())
+            _strip_namespace(self.properties.unit),
+            self.properties.scaling_factor,
```

Comment:
why not _strip_namespace for scaling_factor?

Reply:
As I understand it, scaling factor is a value and not a reference, hence no namespaces. Correct me if that's not the case.